typed-racket-lib/typed-racket/typecheck/tc-let-unit.rkt

```diff
@@ -153,41 +154,43 @@
 ;; For example, it is used to typecheck units and ensure that exported
 ;; variables are exported at the correct types
 (define (tc/letrec-values namess exprs body [expected #f] [check-thunk void])
-  (let* ([names (stx-map syntax->list namess)]
-         [orig-flat-names (apply append names)]
+  (let* ([namess (stx-map syntax->list namess)]
+         [orig-flat-names (apply append namess)]
          [exprs (syntax->list exprs)])
-    (register-aliases-and-declarations names exprs)
+    (register-aliases-and-declarations namess exprs)
     
     ;; First look at the clauses that do not bind the letrec names
     (define all-clauses
-      (for/list ([name-lst names] [expr exprs])
+      (for/list ([name-lst (in-list namess)]
+                 [expr (in-list exprs)])
         (lr-clause name-lst expr)))
 
     (define-values (ordered-clauses remaining)
       (get-non-recursive-clauses all-clauses orig-flat-names))
 
     (define-values (remaining-names remaining-exprs)
-      (for/lists (_1 _2) ([remaining-clause remaining])
+      (for/lists (_1 _2) ([remaining-clause (in-list remaining)])
         (match-define (lr-clause name expr) remaining-clause)
         (values name expr)))
 
     ;; Check those and then check the rest in the extended environment
     (check-non-recursive-clauses
      ordered-clauses
-     (lambda ()
-       (cond
-         ;; after everything, check the body expressions
-         [(null? remaining-names)
-          (check-thunk)
-          (tc-body/check body (and expected (erase-props expected)))]
-         [else
-          (define flat-names (apply append remaining-names))
-          (do-check tc-expr/check
-                    remaining-names
-                    ;; types the user gave.
-                    (map (λ (l) (map tc-result (map get-type l))) remaining-names)
-                    remaining-exprs body expected
-                    check-thunk)])))))
+     (λ ()
+       ;; types the user gave.
```

Comment:
It gave better indentation and I thought perhaps made it clearer that we were entering a sequence of effectful computation where order mattered.